main.go

```diff
@@ -142,13 +142,24 @@ func (handler *BuildHandler) ServeHTTP(
 
 	fmt.Fprintf(topLevelLogger, "running build task for '%s'", repoUrl)
 
-	err := runBuild(
+	err := request.ParseForm()
+	if err != nil {
+		errorMessage := fmt.Sprintf("error parsing request: %s", err)
+		fmt.Fprintf(topLevelLogger, errorMessage)
+		response.WriteHeader(http.StatusBadRequest)
+		response.Write([]byte(errorMessage))
```

Comment:
`topLevelLogger` will already replicate all messages to the `reponse` (see declaration above). So, `WriteHeader` should be called sooner and `reponse.Write` should be removed.